src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,57 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import mate.academy.spring.service.mapper.ResponseDtoMapper;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final MovieSessionService movieSessionService;
+    private final UserService userService;
+    private final ResponseDtoMapper<ShoppingCartResponseDto, ShoppingCart>
+            shoppingCartResponseDtoMapper;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  UserService userService,
+                                  MovieSessionService movieSessionService,
+            ResponseDtoMapper<ShoppingCartResponseDto, ShoppingCart>
+                                      shoppingCartResponseDtoMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+        this.shoppingCartResponseDtoMapper = shoppingCartResponseDtoMapper;
+    }
+
+    @PutMapping("/movie-sessions")
+    public void addToCart(Authentication auth, @RequestParam Long movieSessionId) {
+        UserDetails details = (UserDetails) auth.getPrincipal();
```

Comment:
you don't need UserDetails for getting email, use auth for getting it, the same for other methods